src/test/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMockTest.java

```diff
@@ -371,4 +374,25 @@ public void event(BlockBreakEvent e) throws Exception
 		assertThrowsExactly(EventHandlerException.class, () -> pluginManager.callEvent(new BlockBreakEvent(null, null)));
 	}
 
+	@Test
+	void loadPlugin_InvalidName_DoesntLoad() throws ReflectiveOperationException
+	{
+		// Won't let us create an invalid name.
+		Field nameField = PluginDescriptionFile.class.getDeclaredField("name");
+		nameField.setAccessible(true);
+
+		PluginDescriptionFile sillyName = new PluginDescriptionFile("Name", "1.0.0", TestPlugin.class.getName());
+		nameField.set(sillyName, "Name(with)[other]<chars>!");
+		assertThrows(RuntimeException.class, () -> pluginManager.loadPlugin(TestPlugin.class, sillyName));
+
+		PluginDescriptionFile bukkit = new PluginDescriptionFile("bukkit", "1.0.0", TestPlugin.class.getName());
+		assertThrows(RuntimeException.class, () -> pluginManager.loadPlugin(TestPlugin.class, bukkit));
+
+		PluginDescriptionFile minecraft = new PluginDescriptionFile("minecraft", "1.0.0", TestPlugin.class.getName());
+		assertThrows(RuntimeException.class, () -> pluginManager.loadPlugin(TestPlugin.class, minecraft));
+
+		PluginDescriptionFile mojang = new PluginDescriptionFile("mojang", "1.0.0", TestPlugin.class.getName());
+		assertThrows(RuntimeException.class, () -> pluginManager.loadPlugin(TestPlugin.class, mojang));
+	}
```

Comment:
This should definitely be a parameterized test

Reply:
Right, those exist 😅